python/lsst/ts/m2gui/utility_monitor.py

```diff
@@ -0,0 +1,439 @@
+# This file is part of ts_m2gui.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["UtilityMonitor"]
+
+import re
+
+from . import (
+    SignalUtility,
+    PowerType,
+    TemperatureGroup,
+    DisplacementSensorDirection,
+    get_tor,
+)
+
+
+class UtilityMonitor(object):
+    """Utility monitor to monitor the utility status.
+
+    Attributes
+    ----------
+    signal_utility : `SignalUtility`
+        Signal to report the utility status.
+    power_motor : `dict`
+        Motor power. The unit of voltage is volt and the unit of current is
+        ampere.
+    power_communication : `dict`
+        Communication power. The unit of voltage is volt and the unit of
+        current is ampere.
+    inclinometer_angle : `float`
+        Inclinometer angle in degree.
+    breakers : `dict`
+        Breakers. The key is the breaker name. If the breaker is triggered, the
+        value is True. Otherwise, False.
+    temperatures : `dict`
+        Tempertures. The key is the temperature sensor name. The value is the
+        related temperature in degree C.
+    displacements : `dict`
+        Displacement sensors. The key is the displacement sensor name. The
+        value is the reading in the unit of mm.
+    """
+
+    # Number of digits after the decimal
+    # It is meaningless to have a high accurate value such as x.xxxx shown on
+    # the graphical user interface (GUI) at the most of time. In addition, the
+    # high-frequent update of GUI might drain the limited computer resource.
```

Comment:
It is. But you cannot estimate whoever will need high precision decimals (maybe to convert from some big units into smaller, e.g from A to mA). It's a view thing to format numbers. setText doesn't redraw QLabel if the value is the same. Please drop this constant, round in calls, and use f"{value:.1f}" for the setText call (or provide the custom class to handle formatting).

Reply:
The small fluctuation of voltage, current, and angle are not very important. The move of rigid body is very rare and readings of displacement sensors are just comes from the fluctuation at the most of time. The M2 has no movement such as M1M3 mirror raise. In addition, the EUI will be used by hardware engineers in the engineering mode only. All the telemetry will be published to EFD by M2 CSC. The LOVE will subscribe those data and put into LOVE GUI, which is the main interface for the scientists to use.

In addition, I think it is better to let the `utility_monitor.py` (part of `Model` to decide the values contains the digits to show). It is weird to me that you send a signal to "View" that it will not affect the format. I prefer to save the "Signal" resource and let "View" to show whatever "Model" wants it to show to simplify the business logic.

Thanks!